include/cxx.h

```diff
@@ -235,9 +235,13 @@ class Vec final {
 
   class const_iterator {
   public:
+    using difference_type = ptrdiff_t;
     using value_type = typename std::add_const<T>::type;
+    using pointer = typename std::add_pointer<
+        typename std::add_const<T>::type>::type;
     using reference = typename std::add_lvalue_reference<
         typename std::add_const<T>::type>::type;
+    using iterator_category = std::forward_iterator_tag;
```

Comment:
I think this isn't quite correct because there are requirements on forward iterator not satisfied by this code yet.

https://en.cppreference.com/w/cpp/named_req/ForwardIterator

```cpp
void c_take_rust_vec_shared(rust::Vec<Shared> v) {
  auto it = v.begin();
  (void)it++;
  (void)it->z;
```

```console
tests.cc: In function ‘void tests::c_take_rust_vec_shared(rust::cxxbridge02::Vec<tests::Shared>)’:
tests.cc:195:11: error: no ‘operator++(int)’ declared for postfix ‘++’ [-fpermissive]
  195 |   (void)it++;
      |         ~~^~
tests.cc:196:11: error: base operand of ‘->’ has non-pointer type ‘rust::cxxbridge02::Vec<tests::Shared>::const_iterator’
  196 |   (void)it->z;
      |           ^~
```

Reply:
Implemented forward iterator requirements c0fb2b0